slang-testing/src/test/java/com/sonarsource/slang/testing/TreeAssertTest.java

```diff
@@ -163,6 +165,26 @@ public void notequivalent_failure() {
     assertTree(LITERAL_42).isNotEquivalentTo(new LiteralTreeImpl(null, "42"));
   }
 
+  @Test
+  public void hasdescendant_ok() {
+    assertTree(ABC_PLUS_ABC_PLUS_42).hasDescendant(new LiteralTreeImpl(null, "42"));
+  }
+
+  @Test(expected = AssertionError.class)
+  public void hasdescendant_failure() {
+    assertTree(ABC_PLUS_ABC_PLUS_42).hasDescendant(new LiteralTreeImpl(null, "43"));
+  }
+
+  @Test
```

Comment:
I fail to see the value of these tests. It is clear from the implementation that a tree cannot be both a descendant and not a descendant. If you really want to keep these tests, why not just assert !hasDescendant == hasNotDescendant?

Reply:
I agree, it's clear what they do from the implementation, but they still need to be unit tested :)
I don't think testing `!hasDescendant == hasNotDescendant` would be useful, even less now that they share common method implementation. 
Here, we at least ensure that the assert works as we expect